src/ring_theory/trace.lean

```diff
@@ -97,6 +97,38 @@ begin
   { simp [trace_eq_zero_of_not_exists_basis K H, finrank_eq_zero_of_not_exists_basis_finset H] }
 end
 
+lemma trace_trace_of_basis [algebra S T] [is_scalar_tower R S T]
+  {ι κ : Type*} [fintype ι] [fintype κ]
+  (b : basis ι R S) (c : basis κ S T) (x : T) :
+  trace R S (trace S T x) = trace R T x :=
+begin
+  haveI := classical.dec_eq ι,
+  haveI := classical.dec_eq κ,
+  rw [trace_eq_matrix_trace (b.smul c), trace_eq_matrix_trace b, trace_eq_matrix_trace c,
+      matrix.trace_apply, matrix.trace_apply, matrix.trace_apply,
+      ← finset.univ_product_univ, finset.sum_product],
+  refine finset.sum_congr rfl (λ i _, _),
+  simp only [alg_hom.map_sum, smul_left_mul_matrix, finset.sum_apply,
+      -- The unifier is not smart enough to apply this one by itself:
+      finset.sum_apply i _ (λ y, left_mul_matrix b (left_mul_matrix c x y y))]
+end
+
+lemma trace_comp_trace_of_basis [algebra S T] [is_scalar_tower R S T]
+  {ι κ : Type*} [fintype ι] [fintype κ]
+  (b : basis ι R S) (c : basis κ S T) :
+  (trace R S).comp ((trace S T).restrict_scalars R) = trace R T :=
+by { ext, rw [linear_map.comp_apply, linear_map.restrict_scalars_apply, trace_trace_of_basis b c] }
+
+lemma trace_trace [algebra K T] [algebra L T] [is_scalar_tower K L T]
```

Comment:
```suggestion
@[simp] lemma trace_trace [algebra K T] [algebra L T] [is_scalar_tower K L T]
```

Reply:
This gives a linting error since it conflicts with `@[simps] def trace`. So I removed the `simps` from `trace` and added this one.